roles/bootstrap-os/tasks/bootstrap-redhat.yml

```diff
@@ -79,6 +81,7 @@
     state: "{{ 'enabled' if (rhel_enable_repos | default(True) | bool) else 'disabled' }}"
   when:
     - ansible_distribution_major_version == "7"
+    - (not rh_subscription_status.changed) or ('rc' in rh_sub_orgid_register_status and rh_sub_orgid_register_status.rc == 0) or ('rc' in rh_sub_username_register_status and rh_sub_username_register_status.rc == 0)
```

Comment:
I don't have a huge knowledge of rhel/rhel in kubespray but this checks looks a bit suspicious. Maybe you could use `.failed` and possibly `.skipped` instead? I am not sure about the `not rh_subscription_status.changed` which would skip it if you run the playbook a second time (kinda bad for idempotency) and it seems better perhaps to use failed rather than the condition with `rc`

Reply:
Thanks for your suggestions, I updated it. PTAL